tests/extent_tests.cpp

```diff
@@ -6,10 +6,30 @@
 #include "common/geom.hpp"
 #include "common/helpers/eigen_helpers.hpp"
 #include "ear/bs2051.hpp"
+#include "eigen_utils.hpp"
 #include "object_based/extent.hpp"
 
 using namespace ear;
 
+TEST_CASE("test_AngleToWeight") {
+  for (int start_angle = 0; start_angle <= 180; start_angle += 2) {
+    int end_angle = start_angle + 10;
+    AngleToWeight atw(radians(start_angle), radians(end_angle));
```

Comment:
I get a division by zero running this test, I think the radians template function is casting back to int as the angles are specified as int so 0, 10 degrees winds up as 0, 0 radians.

Reply:
Could just fix the test, but something like https://godbolt.org/z/YW8vb8 might be better unless integer arithmetic is the right thing to do somewhere.

Reply:
Ah sorry, i should have said that i was working on this.

Thanks for spotting it; that test was completely broken.

I just added an `enable_if` to check for a floating point type. Yours is a bit nicer to use sometimes, but this way feels a bit simpler and is less likely to hide other issues.

I added an assert to check that the division by zero isn't happening -- that shouldn't come up in actual usage though.

Reply:
Yeah your enable_if solution is probably better, a compile time nope is unambiguous where as promoting ints could hide a problem elsewhere.